ts/features/idpay/unsubscription/screens/UnsubscriptionConfirmationScreen.tsx

```diff
@@ -0,0 +1,159 @@
+  [
+    {
+      title: I18n.t("idpay.unsubscription.checks.1.title"),
+      subtitle: I18n.t("idpay.unsubscription.checks.1.content")
+    },
+    {
+      title: I18n.t("idpay.unsubscription.checks.2.title"),
+      subtitle: I18n.t("idpay.unsubscription.checks.2.content")
+    },
+    {
+      title: I18n.t("idpay.unsubscription.checks.3.title"),
+      subtitle: I18n.t("idpay.unsubscription.checks.3.content")
+    },
+    {
+      title: I18n.t("idpay.unsubscription.checks.4.title"),
+      subtitle: I18n.t("idpay.unsubscription.checks.4.content")
+    }
+  ];
+
+const UnsubscriptionConfirmationScreen = () => {
+  const machine = useUnsubscriptionMachineService();
+
+  const isLoading = useSelector(machine, isLoadingSelector);
+  const initiativeName = useSelector(machine, selectInitiativeName);
+
+  const checks = useConfirmationChecks(unsubscriptionChecks.length);
+
+  const handleClosePress = () =>
+    machine.send({
+      type: "EXIT"
+    });
+
+  const handleConfirmPress = () => {
+    machine.send({
+      type: "CONFIRM_UNSUBSCRIPTION"
+    });
+  };
+
+  const handleCheckToggle = (index: number) => checks.toggle(index);
+
+  const closeButton = (
+    <TouchableDefaultOpacity
+      onPress={handleClosePress}
+      accessible={true}
+      accessibilityLabel={I18n.t("global.buttons.back")}
+      accessibilityRole={"button"}
+    >
+      <IconFont name={"io-close"} style={{ color: IOColors.bluegrey }} />
+    </TouchableDefaultOpacity>
+  );
+
+  const confirmModal = useIOBottomSheetModal(
+    <Body>{I18n.t("idpay.unsubscription.modal.content")}</Body>,
+
+    I18n.t("idpay.unsubscription.modal.title", { initiativeName }),
+    250,
+
+    <FooterWithButtons
+      type="TwoButtonsInlineHalf"
+      leftButton={{
+        onPress: () => {
+          confirmModal.dismiss();
+          handleConfirmPress();
+        },
+        block: true,
+        bordered: true,
+        title: I18n.t("idpay.unsubscription.button.continue"),
+        danger: true
+      }}
+      rightButton={{
+        onPress: () => {
+          confirmModal.dismiss();
+        },
+        block: true,
+        bordered: true,
+        title: I18n.t("global.buttons.cancel"),
+        labelColor: IOColors.blue
+      }}
+    />
+  );
+
+  const body = (
+    <SafeAreaView style={IOStyles.flex}>
+      <ScrollView>
+        <ContentWrapper>
+          <VSpacer size={16} />
+          <H1>{I18n.t("idpay.unsubscription.title", { initiativeName })}</H1>
+          <VSpacer size={16} />
+          <Body>{I18n.t("idpay.unsubscription.subtitle")}</Body>
+          <VSpacer size={16} />
+          {unsubscriptionChecks.map((item, index) => (
+            <UnsubscriptionCheckListItem
+              key={index}
+              title={item.title}
+              subtitle={item.subtitle}
+              checked={checks.values[index]}
+              onValueChange={() => handleCheckToggle(index)}
```

Comment:
You are right! Initially `useConfirmationChecks` was designed to only toggle a bool value in a list. In https://github.com/pagopa/io-app/pull/4462/commits/5a263d04d38b68725c26394e4ed537746e802a15 I have changed the APIs and now, instead of a `toggle(index)` method, it has a `setValue(index,value)` value. This should allow a better sync between hook and UI.